src/lib/rule2F.ts

```diff
@@ -226,7 +226,10 @@ function matchesRole(elem: HTMLElement, roleType: RoleType): boolean {
  */
 function allowsNameFromContent(elem: HTMLElement, context: Context): boolean {
   // focasable elements should allow name from content.
-  if (elem.hasAttribute('tabindex')) {
+  if (
+    matchesRole(elem, NEVER_NAME_FROM_CONTENT) &&
+    elem.hasAttribute('tabindex')
+  ) {
     return true;
   }
```

Comment:
Does moving this to L252 work? Or does L240 catch some elements?

```typescript
  if (matchesRole(elem, NEVER_NAME_FROM_CONTENT)) {
    // focasable elements should allow name from content.
    return elem.hasAttribute('tabindex');
  }
```

Reply:
At L252 it wouldn't overwrite list 2 roles, so I think this check has to come first, right?